vnext/Microsoft.ReactNative/Modules/AlertModule.cpp

```diff
@@ -71,6 +74,22 @@ void Alert::ProcessPendingAlertRequests() noexcept {
     }
   }
 
+  // Workaround XAML bug with ContentDialog and dark theme:
+  // https://github.com/microsoft/microsoft-ui-xaml/issues/2331
+  dialog.Opened([useXamlRootForThemeBugWorkaround](winrt::IInspectable const &sender, auto &&) {
```

Comment:
this only repros in islands, right? if so, we should only do this dance if the app is an islands app, there are some helpers we use elsewhere for detecting this

Reply:
It's not only Islands. You can see the bug (and fix) in UWP in screenshots above.